src/main/scala/io/movies/interpreters/RepositoryImpl.scala

```diff
@@ -0,0 +1,37 @@
+package io.movies.interpreters
+
+import cats.effect._
+import doobie._
+import doobie.hikari._
+import doobie.implicits._
+import io.movies.algebras.Repository
+import io.movies.model.{Movie, RegisteredMovie}
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+
+object RepositoryImpl {
+
+  //no se pasa el recurso para que no se crea/destruya cada vez. no es viable en el caso de conexiones a BBDD
+  def doobie[F[_]](transactor: HikariTransactor[F])(implicit S: Sync[F], logger: SelfAwareStructuredLogger[F]) : F[Repository[F]] =
+    S.pure(new Repository[F] {
+
+    override def addMovie(movie: Movie): F[Int] = {
```

Comment:
Instead of returning an `Int` value representing the number of affected rows, you could return the recently-created movie. Please, check out the [doobie documentation](https://tpolecat.github.io/doobie/docs/07-Updating.html#retrieving-results) to see how to do it.

Reply:
See https://github.com/gutiory/MoviesServer/pull/1/commits/d47f7920851309fa4527476bd93715bc55abccfe and https://github.com/gutiory/MoviesServer/pull/1/commits/3d756dcf8d76e19baabd1e292fcba68df7216360